test_queue_job/tests/test_delay_mocks.py

```diff
@@ -268,6 +268,22 @@ def test_trap_jobs_perform(self):
             self.assertEqual(logs[2].message, "test_trap_jobs_perform graph 3")
             self.assertEqual(logs[3].message, "test_trap_jobs_perform graph 1")
 
+    def test_trap_jobs_prepare_context(self):
+        # pylint: disable=context-overridden
+        with trap_jobs() as trap:
+            model1 = self.env["test.queue.job"].with_context({"config_key": 42})
```

Comment:
[`config_key` is not part of keys to forward](https://github.com/OCA/queue/blob/14.0/test_queue_job/models/test_models.py#L42), isn't that your problem ?

I suppose you should rely on queue_job_context module: https://github.com/OCA/queue/blob/14.0/queue_job_context/models/base.py#L8 (which this test_queue_job to not depends on)

Reply:
I believe the issue that is fixed here is that the context is now also filtered of trapped jobs, as is the case in queued jobs in production environments. I agree with the original submitter that this is an improvement because it makes the test simulate production environments more realistically.

Reply:
Test model `test.queue.job` redefines the whitelisted keys of the context:
https://github.com/florentx/queue/blob/14.0_trap_context/test_queue_job/models/test_models.py#L42-L43

Reply:
Thanks for clarifying! It would be good to have it documented in the test's docstring that this method is implemented by the test model.